platforms/posix/src/px4/common/main.cpp

```diff
@@ -174,6 +177,25 @@ int main(int argc, char **argv)
 		return client.process_args(argc, (const char **)argv);
 
 	} else {
+#if defined(__PX4_LINUX)	// Linux specific
```

Comment:
You could put this behind `_POSIX_MEMLOCK` instead. Then handle the non-root case failure silently (SITL, etc).

Reply:
Small question: How to handle SITL case? I cannot find a way to properly catch the SITL start.

Reply:
Why handle it at all? The vast majority of time SITL runs as a regular user, if you happen to run it as root you get the same behavior.

Reply:
My current thinking is that SITL and PX4 running on a Linux board are basically the same thing but with simulation drivers instead of real.

Reply:
Notify nothing if px4 runs with regular user, and only mlock if is started by root? I think px4 should be started with mlock by default, so if system supports that but not started with, there should be a warning.

Reply:
I think we treat it the same as real time threads. Maybe a generic warning that PX4 should be running as root for real flight?

Reply:
Warn (or even error) if not running at root unless built with `ENABLE_LOCKSTEP_SCHEDULER` would cover all the cases.